app/controllers/posts_controller.rb

```diff
@@ -1,8 +1,7 @@
 class PostsController < ApplicationController
   def index
     @user = User.find_by(id: params[:user_id])
-
-    @posts = @user.posts.includes(:comments)
+    @posts = Post.includes(:author).paginate(page: params[:page], per_page: 3)
   end
 
   def show
```

Comment:
- [ ] @Enning94 and @WilhelmK109 great work trying to fix the N + 1 problem, however the project requirements indicated that the problem should be fixed when fetching all user's posts and comments, to make the query faster , I can see you had already fixed the problem correctly then changed to new code kindly consider using ` @posts = @user.posts.includes(:comments)` in your index method and remove `  @post = Post.includes(:author, comments: :author).find(params[:id])` in your show method.
![image](https://github.com/Enning94/Blog_App/assets/61408860/97f6d41d-441f-43c7-8bc3-cabf09da3643)